src/cpp/common/alarm.cc

```diff
@@ -31,21 +31,17 @@
  */
 
 #include <grpc++/alarm.h>
-#include <grpc++/completion_queue.h>
 #include <grpc++/impl/grpc_library.h>
-#include <grpc/grpc.h>
 
 namespace grpc {
 
 static internal::GrpcLibraryInitializer g_gli_initializer;
-Alarm::Alarm(CompletionQueue* cq, gpr_timespec deadline, void* tag)
-    : tag_(tag),
-      alarm_(grpc_alarm_create(cq->cq(), deadline, static_cast<void*>(&tag_))) {
+
+Alarm::~Alarm() {
   g_gli_initializer.summon();
```

Comment:
Hold the phone... now I'm a little bothered. I think indeed that this case is only being saved because of CQ. The C++ standard only guarantees that a static object will gets its constructor called before the first use of an object or function defined in its translation unit. In this case, its translation unit is alarm.cc and the first use of a function defined there is the destructor. So, g_gli might not get constructed until the alarm is destructed if you're relying on this. You don't notice this happening because CQ is already created (or because the compiler is not as aggressive as the standard allows). CQ is actually doing the initialization.